pkg/providers/tinkerbell/validate.go

```diff
@@ -27,11 +27,22 @@ func validateOsFamily(spec *ClusterSpec) error {
 		if spec.MachineConfigs[groupRef.Name].OSFamily() != controlPlaneOsFamily {
 			return fmt.Errorf("worker node group osFamily cannot be different from control plane osFamily")
 		}
+		if group.UpgradeRolloutStrategy != nil {
+			if spec.MachineConfigs[groupRef.Name].OSFamily() != v1alpha1.DefaultOSFamily && group.UpgradeRolloutStrategy.Type == "InPlace" {
+				return fmt.Errorf("InPlace upgrades are only supported on the Ubuntu OS family")
+			}
+		}
 	}
 
 	if controlPlaneOsFamily != v1alpha1.Bottlerocket && spec.DatacenterConfig.Spec.OSImageURL == "" && spec.ControlPlaneMachineConfig().Spec.OSImageURL == "" {
 		return fmt.Errorf("please use bottlerocket as osFamily for auto-importing or provide a valid osImageURL")
 	}
+
+	if spec.ControlPlaneConfiguration().UpgradeRolloutStrategy != nil {
+		if controlPlaneOsFamily != v1alpha1.DefaultOSFamily && spec.ControlPlaneConfiguration().UpgradeRolloutStrategy.Type == "InPlace" {
```

Comment:
While we check for the CP and Worker rollout strategy types separately, do we also want to check if one is specified as `InPlace` other cannot be rolling.? I don't think our controllers are designed in such a way but this will prevent unnecessary complications.

Reply:
Sure that makes sense. There is a separate task for fully validating this feature, so I can include this additional check there